src/UIHandler.cpp

```diff
@@ -0,0 +1,345 @@
+    return 0;
+}
+
+void UIWindow::DrawEverything()
+{
+    werase( m_wf_win );
+    m_panel->DrawEverything( m_wf_win, { 0, 0 } );
+}
+
+std::vector<std::shared_ptr<UIPanel>> UIPaddingPanel::GetChild() const
+{
+    return m_childPanels;
+}
+
+void UIPaddingPanel::AddChild( std::shared_ptr<UIPanel> panel )
+{
+    if( !m_childPanels.empty() ) {
+        DebugLog( D_ERROR, DC_ALL ) << "Only supports one panel";
+        return;
+    }
+    m_childPanels.push_back( panel );
+}
+
+void UIPaddingPanel::RemoveChild( size_t index )
+{
+    if( index != m_childPanels.size() - 1 ) {
+        m_childPanels[index] = std::move( m_childPanels.back() );
+    }
+    m_childPanels.pop_back();
+}
+
+point UIPaddingPanel::RequestedSize( Sizes sizes )
+{
+    point size;
+
+    if( m_drawBorder )
+        size += { 2, 2 };
+
+    if( m_childPanels.empty() ) {
+        return size;
+    }
+
+    auto paneSize = m_childPanels[0]->RequestedSize( sizes );
+    size += paneSize;
+
+    return size;
+}
+
+// We are a simple border!
+int UIPaddingPanel::SetSize( point size )
+{
+    m_thisSize = size;
+
+    if( m_childPanels.empty() ) {
+        return 0;
+    }
+    auto newSize = size;
+
+    if( m_drawBorder )
+        newSize -= { 2, 2 };
+
+    m_childPanels[0]->SetSize( newSize );
+    return 0;
+}
+
+void UIPaddingPanel::DrawEverything( WINDOW *wf_win, point offset )
+{
+
+    if( m_drawBorder ) {
+        UIUtils::DrawBorder( wf_win, offset, m_thisSize );
+    }
+
+    if( !m_childPanels.empty() ) {
+        point offset;
+
+        if( m_drawBorder )
+            offset += { 1, 1 };
+
+        m_childPanels[0]->DrawEverything( wf_win, offset );
+    }
+
+    m_lastSize = m_thisSize;
+}
+
+
+UIPaddingPanel::UIPaddingPanel( bool drawBorder )
+{
+    m_drawBorder = drawBorder;
+}
+
+std::vector<std::shared_ptr<UIPanel>> UITabPanel::GetChild() const
+{
+    return m_childPanels;
+}
+
+void UITabPanel::AddChild( std::shared_ptr<UIPanel> panel )
+{
+    AddChild( std::to_string( m_childPanels.size() ), panel );
+}
+
+void UITabPanel::AddChild( std::string name, std::shared_ptr<UIPanel> panel )
+{
+    m_childPanels.push_back( panel );
+    m_childPanelNames.push_back( name );
+}
+
+void UITabPanel::RemoveChild( size_t index )
+{
+    if( index != m_childPanels.size() - 1 ) {
+        m_childPanels[index] = std::move( m_childPanels.back() );
+    }
+    m_childPanels.pop_back();
+
+    if( index != m_childPanelNames.size() - 1 ) {
+        m_childPanelNames[index] = std::move( m_childPanelNames.back() );
+    }
+    m_childPanelNames.pop_back();
+}
+
+point UITabPanel::RequestedSize( Sizes sizes )
+{
+    point size;
+
+    size += { 0, 2 };
+
+    if( m_drawBorder )
+        size += { 2, 2 };
+
+    if( m_childPanels.empty() ) {
+        return size;
+    }
+
+    auto paneSize = m_childPanels[m_currentTab]->RequestedSize( sizes );
+    size += paneSize;
+
+    int len = 2;
+
+    for( auto tx : m_childPanelNames ) {
+        len += utf8_width( tx ) + 6;
```

Comment:
A commit about the meaning of this magic number would be helpful. Also why start with `len = 2`?